src/runtime/ndarray.cc

```diff
@@ -15,6 +21,50 @@ extern "C" void NDArrayDLPackDeleter(DLManagedTensor* tensor);
 namespace dgl {
 namespace runtime {
 
+#ifndef _WIN32
```

Comment:
Similarly can we have these in another source file (like `src/runtime/shared_memory.cpp`)?